File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {

        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++)
        {
            if (hashmap.count(nums[i]) > 0)
                hashmap[nums[i]]++;
            else
                hashmap[nums[i]] = 1;
        }

        int k = 0;
        for (int i = 0; i < 3; i++)
        {
            for (int j = 0; j < hashmap[i]; j++)
            {
                nums[k++] = i;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l, i, r = 0, 0, len(nums) - 1
        while i <= r:
            if nums[i] == 0:
                nums[l], nums[i] = nums[i], nums[l]
                i += 1
                l += 1
            elif nums[i] == 2:
                nums[i], nums[r] = nums[r], nums[i]
                r -= 1
            else: # nums[i] == 1
                i += 1
